install/install.sh

```diff
@@ -45,328 +45,362 @@ error() {
-    exit 1
-  fi
+unpack() {
+  archive="$1"
+  dir=$(dirname "$1")
+
+  case "$archive" in
+    *.tar.gz)
+      if [ -n "$verbose" ]; then
+        flags="-v"
+      else
+        flags=
+      fi
+      tar "$flags" -xzf "$archive" -C "$dir"
+      return 0
+      ;;
+    *.zip)
+      if [ -z "$verbose" ]; then
+        flags="-qq"
+      else
+        flags=
+      fi
+      UNZIP="$flags" unzip "$archive" -d "$dir"
+      return 0
+      ;;
+  esac
+
+  error "Unknown package extension."
+  printf "\n" >&2
+  info "This almost certainly results from a bug in this script--please file a"
+  info "bug report at https://github.com/starship/starship/issues"
+  return 1
 }
 
 elevate_priv() {
-  if ! hash sudo 2>/dev/null; then
+  if ! has sudo; then
     error 'Could not find the command "sudo", needed to get permissions for install.'
+    printf "\n" >&2
     info "If you are on Windows, please run your shell as an administrator, then"
     info "rerun this script. Otherwise, please run this script as root, or install"
     info "sudo."
-    exit 1
+    return 1
   fi
+
   if ! sudo -v; then
     error "Superuser not granted, aborting installation"
-    exit 1
+    return 1
   fi
-}
 
-install() {
-  local msg
-  local sudo
-
-  if test_writeable "${BIN_DIR}"; then
-    sudo=""
-    msg="Installing Starship, please wait…"
-  else
-    warn "Escalated permission are required to install to ${BIN_DIR}"
-    elevate_priv
-    sudo="sudo"
-    msg="Installing Starship as root, please wait…"
-  fi
-  info "$msg"
-  fetch_and_unpack "${sudo}"
+  return 0
 }
 
-# Currently supporting:
-#   - win (Git Bash)
-#   - darwin
-#   - linux
-#   - linux_musl (Alpine)
-detect_platform() {
-  local platform
-  platform="$(uname -s | tr '[:upper:]' '[:lower:]')"
+uname_platform() {
+  platform=$(uname -s | tr '[:upper:]' '[:lower:]')
 
   # check for MUSL
-  if [ "${platform}" = "linux" ]; then
+  if [ "$platform" = "linux" ]; then
     if ldd /bin/sh | grep -i musl >/dev/null; then
-      platform=unknown-linux-musl
+      platform="unknown-linux-musl"
+    else
+      platform="unknown-linux-gnu"
     fi
   fi
 
   # mingw is Git-Bash
-  if echo "${platform}" | grep -i mingw >/dev/null; then
-    platform=pc-windows-msvc
+  if echo "$platform" | grep -i mingw >/dev/null; then
+    platform="pc-windows-msvc"
   fi
 
-  if [ "${platform}" = "linux" ]; then
-    platform=unknown-linux-gnu
+  if [ "$platform" = "darwin" ]; then
+    platform="apple-darwin"
   fi
 
-  if [ "${platform}" = "darwin" ]; then
-    platform=apple-darwin
-  fi
+  echo $platform
+}
+
+platform_check() {
+  platform=${platform:-"$(uname_platform)"}
+  case "$platform" in
+    apple-darwin) return 0 ;;
+    pc-windows-msvc) return 0 ;;
+    unknown-linux-musl) return 0 ;;
+    unknown-linux-gnu) return 0 ;;
+  esac
+
+  arch=$(uname_arch)
 
-  echo "${platform}"
+  error "Builds for $platform are not yet available for Starship"
+  printf "\n" >&2
```

Comment:
Why not place this `\n` in the error string?

Reply:
Unfortunately, print modifiers work only inside the first `printf` argument.